test/routs/testSoldierRouts.js

```diff
@@ -43,20 +43,28 @@ describe("Soldiers tests", function () {
         "rank": "segen",
         "limitations": ["yeshiva"]
       }));
-      Http.onreadystatechange = (e) => {
-        if (Http.readyState == 4 && Http.status == 200) {
-          MongoClient.connect(DBurl, function (err, client) {
-            assert.equal(null, err);
-            const db = client.db(dbName);
-            const collection = db.collection('Soldiers')
-            collection.deleteOne({
-              "id": "tt8145643"
-            }, (deleteError) => {
-              client.close();
-              expect(Http.responseText).to.eql('');
-              done();
-            })
-          });
+      Http.onreadystatechange = (connectErr) => {
+        if (connectErr) {
+          done(connectErr);
+        } else {
```

Comment:
If you write `return done(...)` if the previews line you don't needs the else clause and indentation here